hazelcast/src/main/java/com/hazelcast/map/QueryCache.java

```diff
@@ -84,6 +85,13 @@
 @Beta
 public interface QueryCache<K, V> {
 
+    /**
+     * Query cache doesn't make use of Extractors.
+     * But index service usage requires to pass an instance of it.
+     * We use this shared instance for that purpose.
+     */
+    Extractors EMPTY_EXTRACTORS = Extractors.empty();
```

Comment:
Also, it might be unsafe to share a single extractors instance between query caches belonging to different maps since the instance is actually mutable (see Extractors.getterCache).

Reply:
Good catch, moved `EMPTY_EXTRACTORS` to internal interface.

Regarding instance sharing, sharing one instance should be fine since QueryCache doesn't support extractors.

Reply:
`Extractors` encapsulates attribute value reading logic too, it's not necessary to have value extractors configured for a map to activate this logic. See `Extractors.instantiateGetter`, if there is no extractor configured for an attribute, it fallbacks to portable/reflection-based reading.

Reply:
thanks, should be fixed now.